includes/payment-methods/class-wc-stripe-upe-payment-gateway.php

```diff
@@ -834,6 +834,14 @@ private function process_payment_with_deferred_intent( int $order_id ) {
 
 				// Create a payment intent, or update an existing one associated with the order.
 				$payment_intent = $this->process_payment_intent_for_order( $order, $payment_information );
+			} elseif ( $payment_information['is_using_saved_payment_method'] && 'cashapp' === $selected_payment_type ) {
+				// If the payment method is Cash App Pay and the order has no cost, mark the order as paid.
+				$order->payment_complete();
+
+				return [
+					'result'   => 'success',
+					'redirect' => '',
```

Comment:
```suggestion
					'redirect' => $this->get_return_url( $order ),
```

Without a redirect, using a saved cash app token for an order that doesn't need payment, just leaves the customer on the checkout page.